File: ios/App/Sources/AnalyticsPreferences.swift
import Foundation

/// The user's diagnostics choice, persisted locally (mirrors `AppearanceMode`'s
/// `@AppStorage` convention so Settings can bind to it directly).
///
/// Defaults to **enabled**: usage reporting is what tells us which worship
/// features earn their place, and nothing personal is collected (see
/// `CoreKit.AnalyticsTracking`). The opt-out exists because this is a worship app
/// — someone who would rather send nothing at all shouldn't have to uninstall to
/// get that.
enum AnalyticsPreferences {
    static let storageKey = "diagnostics.enabled"

    /// `UserDefaults.bool(forKey:)` is `false` when unset, so register the
    /// default rather than inverting the flag's meaning.
    static func registerDefaults(_ defaults: UserDefaults = .standard) {
        defaults.register(defaults: [storageKey: true])
    }

    static func isEnabled(_ defaults: UserDefaults = .standard) -> Bool {
        defaults.object(forKey: storageKey) as? Bool ?? true
    }
}

File: android/app/src/main/kotlin/com/fatwabot/app/analytics/AnalyticsPreferences.kt
package com.fatwabot.app.analytics

import android.content.Context
import android.content.SharedPreferences

/**
 * The user's diagnostics choice. Persisted locally only — it gates the SDKs
 * themselves, so it can't be a server-config value (we must know it before the
 * first network call, and offline).
 *
 * Defaults to enabled: crash reports are what make a bug fixable, and nothing
 * personal is collected (see [com.fatwabot.core.common.AnalyticsTracking]). The
 * opt-out exists because this is a worship app — a user who would rather send
 * nothing at all should not have to uninstall to get that.
 */
class AnalyticsPreferences(context: Context) {
    private val prefs: SharedPreferences =
        context.getSharedPreferences("fatwabot_diagnostics", Context.MODE_PRIVATE)

    var isEnabled: Boolean
        get() = prefs.getBoolean(KEY_ENABLED, true)
        set(value) = prefs.edit().putBoolean(KEY_ENABLED, value).apply()

    private companion object {
        const val KEY_ENABLED = "diagnostics.enabled"
    }
}
